Apple Pay button disappears on Classic Checkout (3644)

stracker-phil opened the pull request:
### Description

This PR refactors the Apple Pay gateway to use the latest code patterns, which are already implemented for Google Pay.

The only visible difference of this PR should address the following bug: When using Apple Pay as a separate gateway (classic checkout), the payment button disappeared after editing the billing address. It re-appeared after toggling to a different gateway and then selecting Apple Pay again.

### Changes

The main change is extending the `ApplePayButton` class from `PaymentButton`. This base class was introduced for Google Pay to combine reusable gateway logic.

To make the changes easier to review, the migration was done in 40+ commits -- all marked with the 🚧 icon.

AlexP11223 committed:
Do not add pay later button in editor

AlexP11223 committed:
Merge branch 'trunk' into PCP-3535-fix-editor-paypal-buttons-grid

AlexP11223 committed:
Merge branch 'trunk' into PCP-3535-fix-editor-paypal-buttons-grid

AlexP11223 committed:
Fix funding source parameter

AlexP11223 committed:
Merge branch 'trunk' into PCP-3535-fix-editor-paypal-buttons-grid

stracker-phil committed:
🐛 Removed deprecated CSS rule

The correct CSS that is expected and handled by the JS integration is generated by PHP in the class method `ApplePayButton::hide_gateway_until_eligible()`

stracker-phil committed:
🚧 Extend class from PaymentButton base

stracker-phil committed:
🚧 Remove unused methods

stracker-phil committed:
🚧 Update the constructor

stracker-phil committed:
🚧 Remove obsolete functions

stracker-phil committed:
🚧 Remove #isInitialized and obsolete getter

stracker-phil committed:
🚧 Fix code style

stracker-phil committed:
🚧 Implement getters for wrapper & styles

stracker-phil committed:
🚧 Add css-class to style the payment button

Used by the parent class

stracker-phil committed:
🚧 Remove obsolete attributes

Those have moved to the parent class

stracker-phil committed:
🚧 Make remaining attributes private

stracker-phil committed:
🚧 Add new validation & configuration methods

stracker-phil committed:
🚧 Minor cleanup

stracker-phil committed:
💡 Add missing annotation in Google Pay module

stracker-phil committed:
🚧 Migrate button initialization to new workflow

The ContextHandlerFactory previously was used inside the ApplePayButton constructor.

stracker-phil committed:
🚧 Adjust button-manager to new workflow

stracker-phil committed:
♻️ Match button-manager of GooglePay and ApplePay

stracker-phil committed:
♻️ Further streamline button-manager classes

stracker-phil committed:
🚧 Swap local object for official config object

stracker-phil committed:
♻️ Simplify the address extraction logic

stracker-phil committed:
🚧 Remove jQuery from boot.js

stracker-phil committed:
🚧 Update boot.js to same logic as Google Pay

stracker-phil committed:
🚧 Minor refactoring in boot.js

stracker-phil committed:
🚧 Remove unused code

stracker-phil committed:
🔀 Merge branch 'PCP-3771’

# Conflicts:
#	modules/ppcp-googlepay/resources/js/GooglepayManager.js

stracker-phil committed:
🔀 Merge branch 'trunk'

stracker-phil committed:
✨ Apply new namespaced script loading

stracker-phil committed:
♻️ Make internal attributes private

stracker-phil committed:
✨ Add transaction info to ApplePay initialization

stracker-phil committed:
🚧 Implement requireShipping getter

stracker-phil committed:
✨ Streamline button configuraton validation logic

stracker-phil committed:
🐛 Fix incorrect validation rule

stracker-phil committed:
🚧 Remove duplicate getters

stracker-phil committed:
🔥 Remove incorrect validation rule

stracker-phil committed:
♻️ Start ApplePaySession after handlers were added

Apple Pay SDK docs hint that “begin()” must be called after setting up the event handlers.

stracker-phil committed:
🚧 Remove more legacy code

stracker-phil committed:
🚧 Clean up the init logic

stracker-phil committed:
🚧 Migrate eligibility check

stracker-phil committed:
🚧 Migrate the addButton logic

stracker-phil committed:
🐛 Add missing event handler to payment button

stracker-phil committed:
🔀 Merge branch 'trunk'

AlexP11223 committed:
Merge branch 'trunk' into PCP-3535-fix-editor-paypal-buttons-grid

stracker-phil committed:
🔀 Merge branch 'trunk'

# Conflicts:
#	modules/ppcp-applepay/resources/js/ApplepayManager.js
#	modules/ppcp-applepay/resources/js/ApplepayManagerBlockEditor.js
#	modules/ppcp-applepay/resources/js/boot-block.js
#	modules/ppcp-applepay/resources/js/boot.js

stracker-phil committed:
🚧 Add block integration (same as GooglePay)

danielhuesken committed:
Display payment gateways also when the are set to display:none with css and the stylesheet is removed from DOM

danieldudzic committed:
Axo: Add support for shipping locations limiting

danieldudzic committed:
Fix Psalm errors

danieldudzic committed:
Axo: Remove the submit button when Fastlane is disabled

inpsyde-maticluznar committed:
Prevent plugin use for send only countries

stracker-phil committed:
🔀 Merge branch 'trunk'

stracker-phil committed:
🐛 Fix identifyer to disable inline-style

The PaymentButton base class expects a “data-hide-gateway” attribute to make an APM gateway visible

stracker-phil committed:
🎨 Minor code improvement

danieldudzic commented on modules/ppcp-applepay/resources/js/boot-block.js:
```diff
@@ -12,26 +13,22 @@ const ppcpConfig = ppcpData.scriptData;
 const buttonData = wc.wcSettings.getSetting( 'ppcp-applepay_data' );
 const buttonConfig = buttonData.scriptData;
-const namespace = 'ppcpBlocksPaypalApplepay';
+const namespace = 'ppcpBlocksPaypalAppglepay';
```
Typo found :) 
```suggestion
const namespace = 'ppcpBlocksPaypalApplepay';
```

danieldudzic reviewed (approved):
Great effort, @stracker-phil.

Only found one minor typo in the namespace, otherwise code is legit.

I'm pre-approving.

stracker-phil committed:
Update modules/ppcp-applepay/resources/js/boot-block.js

Co-authored-by: Danny Dudzic <d.dudzic@syde.com>

AlexP11223 committed:
Merge branch 'trunk' into PCP-3644-apple-pay-button-disappears-on-classic-checkout

inpsyde-maticluznar committed:
Add SendOnlyCountryNotice to notice array separately

inpsyde-maticluznar committed:
Merge remote-tracking branch 'origin/trunk' into PCP-3790-Prevent-plugin-use-for-Send-only-countries

# Conflicts:
#	modules/ppcp-wc-gateway/src/WCGatewayModule.php

AlexP11223 committed:
Sync PayPal product page button state to Apple/Google, show alerts

Dinamiko committed:
Merge pull request #2705 from woocommerce/PCP-2170_Fix_probs_with_autooptimize

Fix problems with autoptimize plugin (2170)

Dinamiko committed:
Merge pull request #2570 from woocommerce/PCP-3535-fix-editor-paypal-buttons-grid

Do not add pay later button in editor (3535)

stracker-phil committed:
🐛 FIx the button preview in admin settings

stracker-phil committed:
🐛 Fix the Apple Pay button style on front end

stracker-phil committed:
Merge branch 'PCP-3644-apple-pay-button-disappears-on-classic-checkout' of github.com:woocommerce/woocommerce-paypal-payments into _done/PCP-3644-apple-pay-button-disappears-on-classic-checkout

stracker-phil committed:
♻️ Separate eligibility flags from user input

Dinamiko committed:
Merge pull request #2711 from woocommerce/PCP-3715-fastlane-limitation-for-shipping-locations

Axo: Integrate shipping restrictions (3715)

Dinamiko committed:
Merge pull request #2720 from woocommerce/PCP-3504-resolve-duplicate-css-id-place-order

Axo: Remove the submit button when Fastlane is disabled (3504)

stracker-phil committed:
🧑‍💻 Add a Reset-Store action for testing

stracker-phil committed:
👔 Introduce new props in the Redux store

stracker-phil committed:
♻️ Move constants to a central config file

Reason: The constants are also used by the store and are synced to the server. Their use is not isolated to the specific component

stracker-phil committed:
👔 Introduce new hook details

stracker-phil committed:
♻️ Consolidate hook props

stracker-phil committed:
♻️ Extract onboarding state to custom hooks

stracker-phil committed:
♻️ Rename boolean getters/setters for data models

To make the function names more semantic

stracker-phil committed:
👔 Add server-side onboarding profile details

stracker-phil committed:
🔒️ Add array sanitizer for products field

AlexP11223 committed:
Exclude node_modules in psalm

AlexP11223 committed:
Add dummy connect_manual endpoint

danielhuesken committed:
Add GW refund support for Apple Pay

AlexP11223 committed:
Update manual inputs labels for sandbox/live

AlexP11223 committed:
Perform manual connection request

AlexP11223 committed:
Fix psalm

stracker-phil committed:
🎨 Standardize imports, remove unneccesary consts

stracker-phil committed:
🏗️ Add conditional routing to onboarding steps

stracker-phil committed:
♻️ Refactor Onboarding component and step handling

stracker-phil committed:
✨ Add casual selling eligibility check

stracker-phil committed:
🔇 Remove console.log from getSteps function

Dinamiko committed:
Merge pull request #2744 from woocommerce/PCP-3820-connect-store-to-onboarding-pages

Connect Store to "Business" and "Product" onboarding steps (3820)

stracker-phil committed:
Merge branch 'trunk' into PCP-3786-manual-input

Dinamiko committed:
Fix phpcs

Dinamiko committed:
Merge pull request #2747 from woocommerce/PCP-3786-manual-input

Handle manual connection in new onboarding

stracker-phil committed:
🌐 Update supported countries for casual selling

Dinamiko committed:
Fix settings ui merge conflict

Dinamiko committed:
Merge pull request #2753 from woocommerce/fix/settings-ui-merge-conflict

Fix settings ui merge conflict

Dinamiko committed:
Fix js error check if element exist before using it

Dinamiko committed:
Merge pull request #2748 from woocommerce/PCP-3820-casual-selling-country-list

Update country list to determine casual selling availability (3813)

AlexP11223 committed:
Update js deps

Dinamiko committed:
Merge pull request #2755 from woocommerce/PCP-3826-apple-pay-is-missing-from-on-all-pages-and-button-styling-preview-on-standard-payment-tab-and-advanced-processing-tab

Apple Pay button preview missing in Standard payment and Advanced Processing tabs (3826)

Dinamiko committed:
Merge pull request #2746 from woocommerce/PCP-3823_Allow-refunds-for-Apple-Pay

Add GW refund support for Apple Pay

Dinamiko committed:
Merge pull request #2721 from woocommerce/PCP-3790-Prevent-plugin-use-for-Send-only-countries

Prevent plugin use for "Send only" countries (3790)

Dinamiko committed:
Merge pull request #2742 from woocommerce/PCP-2142-google-apple-variants-alerts

Sync the PayPal product page button state to Apple/Google Pay buttons, show alerts

Dinamiko committed:
Merge pull request #2757 from woocommerce/PCP-3627-update-js

Update js deps

stracker-phil committed:
🔀 Merge branch 'PCP-3644’

# Conflicts:
#	modules/ppcp-applepay/resources/js/ApplepayButton.js
